platform/kaniko_cache.go

```diff
@@ -0,0 +1,107 @@
+package platform
+
+import (
+	"context"
+	"fmt"
+	"github.com/kiegroup/container-builder/client"
+	"github.com/kiegroup/container-builder/util/defaults"
+	v08 "github.com/kiegroup/kogito-serverless-operator/api/v1alpha08"
+	"github.com/kiegroup/kogito-serverless-operator/builder"
+	"github.com/kiegroup/kogito-serverless-operator/constants"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func createKanikoCacheWarmerPod(ctx context.Context, client client.Client, platform *v08.KogitoServerlessPlatform) error {
+	// The pod will be scheduled to nodes that are selected by the persistent volume
+	// node affinity spec, if any, as provisioned by the persistent volume claim storage
+	// class provisioner.
+	// See:
+	// - https://kubernetes.io/docs/concepts/storage/persistent-volumes/#node-affinity
+	// - https://kubernetes.io/docs/concepts/storage/volumes/#local
+	// nolint: staticcheck
+	pvcName := constants.DEFAULT_KANIKOCACHE_PVC_NAME
+	if persistentVolumeClaim, found := platform.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoPVCName]; found {
+		pvcName = persistentVolumeClaim
+	}
+
+	var warmerImage string
+	if image, found := platform.Status.BuildPlatform.PublishStrategyOptions[builder.KanikoWarmerImage]; found {
+		warmerImage = image
+	} else {
+		warmerImage = fmt.Sprintf("%s:v%s", builder.KanikoDefaultWarmerImageName, defaults.KanikoVersion)
+	}
+
+	pod := corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Pod",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: platform.Namespace,
+			Name:      platform.Name + "-cache",
+			Labels: map[string]string{
+				"camel.apache.org/component": "kaniko-warmer",
+			},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "warm-kaniko-cache",
+					Image: warmerImage,
+					Args: []string{
+						"--cache-dir=" + builder.KanikoCacheDir,
+						"--image=" + platform.Status.BuildPlatform.BaseImage,
+					},
+					VolumeMounts: []corev1.VolumeMount{
+						{
+							Name:      "kaniko-cache",
+							MountPath: builder.KanikoCacheDir,
+						},
+					},
+				},
+			},
+			// Create the cache directory otherwise Kaniko warmer skips caching silently
```

Comment:
isn't better to just log a debug message instead silently skip the caching?

Reply:
The "silent" behavior is not on the operator side but on Kaniko warmer side that without this directory will skip to warm the cache.